modules/3d/include/opencv2/3d/volume_settings.hpp

```diff
@@ -0,0 +1,71 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+
+
+#ifndef OPENCV_3D_VOLUME_SETTINGS_HPP
+#define OPENCV_3D_VOLUME_SETTINGS_HPP
+
+#include <opencv2/core/cvstd.hpp>
+#include <opencv2/3d/volume.hpp>
+
+namespace cv
+{
+
+enum class VolumeType
+{
+    TSDF = 0,
+    HashTSDF = 1,
+    ColorTSDF = 2
+};
+
+
+class CV_EXPORTS_W VolumeSettings
+{
+public:
+    VolumeSettings();
+    VolumeSettings(VolumeType volumeType);
+    ~VolumeSettings();
+
+    void  setWidth(int val);
+    int   getWidth() const;
+    void  setHeight(int val);
+    int   getHeight() const;
```

Comment:
Please transform `set/get Width/Height` functions into `set/get Integrate Width/Height` and  `set/get Raycast Width/Height`.
The reason is the following: each intrinsics comes with its corresponding image size; therefore since we have separate intrinsics for integrate and raycast we should have separate width and height.